submissions/f3d0t/friends-app/app.js

```diff
@@ -0,0 +1,166 @@
+		this.wrapper.innerHTML = "";
+		this.wrapper.append(...cardsArray.map((card) => card.createCard()));
+	}
+	resetFilters() {
+		this.currentCards = [...this.allCards];
+		this.renderCards();
+	}
+	filterByGender(gender) {
+		if (gender == "male") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard["gender"] == "male");
+		}
+		if (gender == "female") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard["gender"] == "female");
+		}
+		if (gender == "all") {
+			this.currentCards = [...this.allCards];
+		}
+		this.renderCards();
+	}
+	filterBySearch(searchValue) {
+		const filteredArray = this.currentCards.filter((friendCard) => Object.values(friendCard).slice(1).join(" ").toLowerCase().indexOf(searchValue.toLowerCase()) > -1);
+		this.renderCards(filteredArray);
+	}
+	sortByName(sortType) {
+		if (sortType == "az") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "name", "ascending");
+		}
+		if (sortType == "za") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "name", "descending");
+		}
+		this.renderCards();
+	}
+	sortByAge(sortType) {
+		if (sortType == "ascending") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "age", "ascending");
+		}
+		if (sortType == "descending") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "age", "descending");
+		}
+		this.renderCards();
+	}
+}
+
+const sortObjectsByPropertyValue = (arrayOfObjects, key, sortOrder) => {
+	const array = arrayOfObjects;
+	const modifier = sortOrder == "ascending" ? 1 : -1;
+	return array.sort((a, b) => (a[key] > b[key] ? 1 * modifier : a[key] < b[key] ? -1 * modifier : 0));
+};
+
+class FriendCard {
+	constructor(friendData) {
+		this.gender = friendData.gender;
+		this.name = `${friendData.name.first} ${friendData.name.last}`;
+		this.age = friendData.dob.age;
+		this.photo = friendData.picture.large;
+		this.email = friendData.email;
+		this.phone = friendData.cell;
+		this.country = friendData.location.country;
+	}
+
+	createCard() {
+		const card = document.createElement("div");
+		card.classList.add("card");
+		card.innerHTML = `<img class="card__img" src="${this.photo}" alt="${this.name} photo"'>\n
+                          <h3 class="card__name">${this.name}</h3>\n
+                          <span class="card__age">Age: ${this.age}</span>\n
+                          <a class="card__phone" href="tel:+${this.phone.replace(/\D/g, "")}">${this.phone}</a>\n
+                          <a class="card__email" href="mailto:${this.email}">${this.email}</a>\n
+                          <span class="card__country">${this.country}</span>`;
+		return card;
+	}
+}
+
+const dayNightChange = () => {
+	if (!DAY_NIGHT.classList.contains("light")) {
+		document.documentElement.style.setProperty("--light", "#111");
+		document.documentElement.style.setProperty("--dark", "#eee");
+		document.documentElement.style.setProperty("--gray", "#ccc");
```

Comment:
Why you can't do this with pure CSS?

Reply:
I can do It with pure CSS, but is it necessarily? To do this in pure CSS, I should declare both LightTheme and DarkTheme custom properties in :root pseudo selector, then declare --active custom properties at body{} selector with active theme styling, and then override it with changing the class of the body to "dark" with JS. It will contain really much code unlike this approach. Check option 4 in first answer in [this](https://stackoverflow.com/questions/58206867/overriding-root-css-variables-from-inner-scopes) question on stackoverflow.
It is happened because we can't override CSS custom property declared in :root pseudo-element from anywhere of other CSS. Only we can do is to override it at top parent element.

Reply:
> I can do It with pure CSS, but is it necessarily?

Kind of. If something can be done with CSS - it should be done with CSS. General idea is fact that JS for logic and CSS is for styles.

> To do this in pure CSS, I should declare both LightTheme and DarkTheme custom properties in :root pseudo selector, then declare --active custom properties at body{} selector with active theme styling, and then override it with changing the class of the body to "dark" with JS. 

This is how pretty much how CSS states looks like. You declare both states in CSS and then switch them with additional class name.

>  It will contain really more code unlike current approach.

You write code for humans, not for computer. Think about this in another way:

1. It is more declarative
1. You keep everything in one place and if someone will open your file, he will see all values
1. You will keep everything related to styles in CSS
1. And pure CSS is cool! 🔥

Reply:
Okay, I will do this =)